### DISCUSSION TITLE
Configure different package rules for a regexManager

### DISCUSSION THREAD

**Discussion by mathijshoogland** [renovatebot/renovate#16954] [Request Help] [answered]
Hi all, I'm struggling a bit with my renovate config and was wondering if somebody knew if I was doing something wrong. I've got a regex manager that's configured to look for files that match the following path: `^infrastructure/common/.*kustomization.yaml$`, I know want to apply labels and policies based on if the path ends with `prod` or `dev`. I'd expect to following config to work but my labels are not being applied. Am I overseeing something?

```
{
  "extends": ["config:base"],
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "regexManagers": [
    {
      "fileMatch": ["^infrastructure/common/.*kustomization.yaml$"],
      "matchStrings": [
        "registryUrl=(?<registryUrl>.*?) chart=(?<depName>.*?)\\s *value: (?<currentValue>.*)\\s"
      ],
      "datasourceTemplate": "helm"
    }
  ],
  "packageRules": [
    {
      "matchManagers": ["regex"],
      "groupName": "production",
      "matchPaths": ["infrastructure/common/**/overlays/prod"],
      "labels": ["prod"],
      "commitMessagePrefix": ["prod -"]
    },
    {
      "matchManagers": ["regex"],
      "groupName": "production",
      "matchPaths": ["infrastructure/common/**/overlays/dev"],
      "labels": ["prod"],
      "commitMessagePrefix": ["dev -"]
    }
  ]
}
```

**Comment by Gabriel-Ladzaretti** [comment, marked as answer]
I think the problem is with your `matchPaths`, either have a ** wildcard at the end of it, or append the filename to it.

for example having the following config
```json
{
  "packageRules": [
    {
      "groupName": "prod",
      "matchPaths": ["prod/**/sub3/**"],
      "addLabels": [
        "prod"
      ]
    },
    {
      "groupName": "dev",
      "matchPaths": ["dev/**/sub4/Dockerfile"],
      "addLabels": [
        "dev"
      ]
    }
  ]
}
```
results in these two prs:
![image](https://user-images.githubusercontent.com/97394622/182852574-eb9a6ccc-8dee-4c6e-a0d2-b044a2c08a96.png)

[source](https://github.com/ladzaretti/labels-testing)

**Reply by mathijshoogland** [reply]
This indeed did the trick, thanks for your help!
